### DISCUSSION TITLE
Adafruit ESP32 Feather V2 keeps resetting

### DISCUSSION THREAD

**Discussion by NJ-Tom** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#614] [Q&A]
I am trying to use an [Adafruit ESP32 Feather V2](https://www.adafruit.com/product/5400) with a [64x32 Hub75 display](https://www.amazon.com/dp/B0BRBGHFKQ), with the code from Examples - ESP32 HUB75 LED MATRIX PANEL DMA Display - 1_SimpleTestShapes, but the display only flashes two lines, and the Arduino Serial Monitor shows this on every flash:

```
22:58:31.142 -> rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
22:58:31.142 -> configsip: 271414342, SPIWP:0xee
22:58:31.142 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
22:58:31.189 -> mode:DIO, clock div:1
22:58:31.189 -> load:0x3fff0030,len:1184
22:58:31.189 -> load:0x40078000,len:13260
22:58:31.189 -> load:0x40080400,len:3028
22:58:31.189 -> entry 0x400805e4
22:58:32.144 -> ets Jul 29 2019 12:21:46
```

So the resetting is apparently causing the flashing. Any ideas as to why this might be happening?

The Feather is attached to a powered USB hub, and the LED display is powered from a bench supply set to 5 V, so there shouldn't be power supply issues. I am not using any level shifters on the data lines, though. Here is the pinout I am using:

```
Hub75   Cable   Func  GPIO  Feather
Pin #   Color   tion  Pin   PCB Label
    1   Brown   R1    25    A1
    2   Red     G1    26    A0
    3   Orange  B1    27    27
    4   Yellow  GND         GND
    5   Green   R2    14    14
    6   Blue    G2    12    12
    7   Violet  B2    13    13
    8   Grey    E     -1    N.C.
    9   White   A     23    SDA
   10   Black   B     19    MI
   11   Brown   C      5    SCK
   12   Red     D     17    TX
   13   Orange  CLK   16    RX
   14   Yellow  LAT    4    A5
   15   Green   OE    15    15
   16   Blue    GND         N.C.
```

I am not set on using that specific Feather; any one that supports Bluetooth is fine. It would be nice to stick with a Feather though, as it is a standard and includes a battery charger circuit.

**Comment by Lukaswnd** [comment]
The Feather V2 has 2MB of PSRAM. This is connected to the gpio pins 16 and 17 (and some more). So these pins are used twice, which might lead to this error. Try replacing 16 and 17 with 32 and 33.

If you error still occurs, try changing the Flashmode to DI or DIO. And test all combinations of it. 
Here a little explation:
If I see it right on the back of the feather V2, RX and TX are gpio 6 and 7. 
These are usually connected to the Flash and the external PSRAM. And GPIOs 1 and 3 are usually RX and TX. I see no difference in the mounted esp32-pico-mini-2. Using them would require to half the flash and psram bandwith. So why did Adafruit do that, if there are other GPIO available to break out on the feather board???

Please inform us, when something worksout.
